### PR TITLE
Improved tracking of integrated heat sink changes during refit

### PR DESCRIPTION
**Pull request by RexPearce**
Fix for #630

### CODE REVIEW

**Review by Windchild292 — COMMENTED** [review]
Just some formatting comments first. Anything relating to refit requires more thought to review because of how the system is done.

**Comment by RexPearce** [comment]
I split the changes into multiple commits to make it easier to follow.

94523d9a51ce8027b5c00976cbeb4c59ada89040 is a bugfix to the original code. The original code was adding HeatSink as Parts to the shoppingList which caused them to be later discarded in .acquireParts(), preventing them from being ordered from the market. With this change, new integrated heat sinks are ordered from the market as expected, then added to the unit being refitted.

However, we don't want actually these heat sink parts added to the unit - they're supposed to represent heat sinks not tracked via Parts. Adding these parts to the unit causes two problems: (i) we bloat units with useless parts and (ii) if this 'bloated' unit is refit again, those extra heat sinks are added back to warehouse resulting in more free/unaccounted heat sinks (the problem we're trying to solve).

1bd5efc05ca97a538a068f3d324daa0dea1d6745 solves the second problem by stripping these extra heat sinks from the unit in the final stage of refit, after the market bookkeeping is complete. It detects them by looking for heat sinks attached to the unit without an assigned location.

518eff0aabb071820c8226d20465dbc5ff7c6050 fixes another problem where the warehouse was never checked for spare heat sinks before ordering from the market.

45f407c020714ed45a4a98d068aa42ea6350b5e3 fixes a couple problems with the first cleanup routine (i) Aero units weren't cleaned up because they use AeroHeatSinks instead of HeatSinks, and (ii) the extra heatsinks in tanks were causing Utilities.unscrambleEquipmentNumbers() to log errors. So now heat sinks are stripped from Tanks and Aeros before unscrambleEquipmentNumbers() is called, and Mechs are done afterwards so the locations are set properly.

**Comment by Windchild292** [comment]
Sorry, badly worded on my part. I meant that changes to this system from anyone can cause unintended issues because of how it was originally coded, and thus any changes can take a while to think over. However, that breakup makes it easier to review.

Furthermore, may I suggest joining us on our Slack? It is easier to reach any of the team and to ask any questions you might find there.

File: MekHQ/src/mekhq/campaign/parts/Refit.java [hunk 1]
```diff
@@ -881,14 +878,16 @@ public void calculate() {
                     oldIntegratedHS.add(oldHS.clone());
                 }
                 for (int i = 0; i < newCount - oldCount; i++) {
-                    newIntegratedHS.add(oldHS.clone());
+                    // Heat sink added for supply chain tracking purposes and removed from refit later
+                    newIntegratedHS.add(newHS.clone().getMissingPart());
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/parts/Refit.java:882]: Can't this just be newHS.getMissingPart() ?

**Review by Windchild292 — APPROVED** [review]
Approved, with the one comment remaining.

**Comment by RexPearce** [comment]
No worries, just trying to save you some time on the review, and explain my reasoning in case I made any mistakes.

(I actually joined the Discord awhile back under a different name, but I haven't used it in awhile. I'll see if I can find the account details.)

**Pull request merged**